scripts/core/editor3/components/toolbar/index.tsx

```diff
@@ -177,15 +183,49 @@ class ToolbarComponent extends React.Component<IProps, IState> {
             editorState,
         } = this.props;
 
-        const has = (opt) => editorFormat.indexOf(opt) > -1;
+        const has = (opt: RICH_FORMATTING_OPTION) => editorFormat.indexOf(opt) > -1;
         const showPopup = (type) => (data) => this.props.showPopup(type, data);
         const cx = classNames({
             'Editor3-controls': true,
             'floating-toolbar': floating,
             disabled: disabled && activeCell === null,
         });
 
-        if (activeCell == null) {
+        if (activeCell != null) {
+            return (
+                <TableControls
+                    className={cx}
+                    tableKind={activeCell.additional.tableKind}
+                    editorFormat={(() => {
+                        switch (activeCell.additional.tableKind) {
+                        case 'table':
+                            return editorFormat;
+                        case 'multi-line-quote':
+                            return editorFormat.filter((option) => option !== 'quote');
+                        case 'custom-block': {
+                            const vocabulary = sdApi.vocabularies.getAll().get(activeCell.additional.vocabularyId);
+
+                            if (vocabulary.field_type !== 'editor-block') {
+                                throw new Error();
+                            }
+
+                            const availableOptions = new Set(getFormattingOptionsForTableLikeBlocks());
+
+                            const vocabularyValues = (
+                                vocabulary.field_options.formatting_options ?? []
+                            ) as Array<RICH_FORMATTING_OPTION>;
+
+                            return vocabularyValues.filter(
+                                (option) => availableOptions.has(option as RICH_FORMATTING_OPTION),
+                            );
+                        }
+                        default:
+                            assertNever(activeCell.additional);
+                        }
+                    })()}
+                />
+            );
+        } else {
```

Comment:
Can you improve nesting for this whole block and inside it? Looks like we can at least avoid using else and get one level up. I think that's fine here since we only have one condition, so it doesn't sacrifice readability